src/ut/handlers_test.cpp

```diff
@@ -633,240 +632,154 @@ class HandlersTest : public testing::Test
-    .WillRepeatedly(DoAll(SetArgReferee<0>(db_regstate), SetArgReferee<1>(db_ttl)));
-  EXPECT_CALL(mock_op, get_charging_addrs(_)).Times(AtLeast(1))
-    .WillRepeatedly(SetArgReferee<0>(NO_CHARGING_ADDRESSES));
-
-  // This isn't a new biding, so add IMPI to the list of associated IMPIs.
-  std::vector<std::string> associated_identities = ASSOCIATED_IDENTITIES;
-  associated_identities.push_back(IMPI);
-  EXPECT_CALL(mock_op, get_associated_impis(_)).Times(AtLeast(1))
-    .WillRepeatedly(SetArgReferee<0>(associated_identities));
-
-  // A Server-Assignment-Request should be generated for this
-  // test. Check for it, and check that its properties are as expected.
-  EXPECT_CALL(*_mock_stack, send(_, _, 200))
-    .Times(1)
-    .WillOnce(WithArgs<0,1>(Invoke(store_msg_tsx)));
-
-  t->on_success(&mock_op);
-
-  ASSERT_FALSE(_caught_diam_tsx == NULL);
-  Diameter::Message msg(_cx_dict, _caught_fd_msg, _mock_stack);
-  Cx::ServerAssignmentRequest sar(msg);
-  EXPECT_TRUE(sar.get_str_from_avp(_cx_dict->DESTINATION_REALM, test_str));
-  EXPECT_EQ(DEST_REALM, test_str);
-  EXPECT_TRUE(sar.get_str_from_avp(_cx_dict->DESTINATION_HOST, test_str));
-  EXPECT_EQ(DEST_HOST, test_str);
-  EXPECT_EQ(IMPI, sar.impi());
-  EXPECT_EQ(IMPU, sar.impu());
-  EXPECT_TRUE(sar.server_name(test_str));
-  EXPECT_EQ(SERVER_NAME, test_str);
-  EXPECT_TRUE(sar.server_assignment_type(test_i32));
-  if (use_wildcard && expect_wc_on_sar)
+  // Template for testing behaviour when there's wildcard identities.
+  //
+  // The basic flow is:
+  //  - Recieve a HTTP request - this may or may not have wildcard information
+  //  - Look up the IMPU/wildcard IMPU from the request in the cache - this
+  //      never finds the user
+  //  - Send a SAR to the HSS
+  //  - Receive an SAA back from the HSS - this always returns the error
+  //      DIAMETER_ERROR_IN_ASSIGNMENT_TYPE. In the error case, it may also
+  //      return wildcard identity information.
+  //      Depending on the UT, we then either:
+  //        - Finish processing, as we're in an error case
+  //        - Lookup in the cache with a new identity and find the information
+  //        - Lookup in the cache with a new identity, don't find the information,
+  //          and send a SAR again. In this case we just timeout the second SAR,
+  //          as we're not interested in any more SAR processing for this UT.
+  //  - Send a HTTP response
+  //
+  // Note - there are no tests for having a wildcard and no external HSS, as this
+  // isn't a supported configuration.
+  void reg_data_template_with_wildcard(std::string reqtype,          // Reqtype on the HTTP request
+                                       std::string initial_wildcard, // Wildcard set on the HTTP request
+                                       std::string wildcard_on_saa,  // Wildcard returned on the SAA
+                                       bool found_subscriber)        // Whether the subscriber was found in the cache
```

Comment:
comment should make it clearer that this means finding the subscriber on the second loop - not the first